server/controllers/stock/core.js

```diff
@@ -269,3 +281,231 @@ function stockManagementProcess(inventories) {
     return inventory;
   });
 }
+
+/**
+ * @function getStockConsumption
+ *
+ * @description returns the monthly (periodic) stock consumption (CM)
+ *
+ * @param {array} periodIds
+ */
+function getStockConsumption(periodIds) {
+  const sql = `
+    SELECT SUM(s.quantity) AS quantity, BUID(i.uuid) AS uuid, i.text, i.code, d.text
+    FROM stock_consumption s
+    JOIN inventory i ON i.uuid = s.inventory_uuid 
+    JOIN depot d ON d.uuid = s.depot_uuid
+    JOIN period p ON p.id = s.period_id
+    WHERE p.id IN (?) 
+    GROUP BY i.uuid, d.uuid
+  `;
+  return db.exec(sql, [periodIds]);
+}
+
+/**
+ * @function getStockConsumptionAverage
+ *
+ * @description returns average of stock consumption (CMM)
+ *
+ * @param {number} periodId - the base period
+ *
+ * @param {number} periodDate - a date for finding the correspondant period
+ *
+ * @param {number} numberOfMonths - the number of months for calculating the average
+ */
+function getStockConsumptionAverage(periodId, periodDate, numberOfMonths) {
+  const baseAvgNumberOfMonths = numberOfMonths || BASE_NUMBER_OF_MONTHS;
+
+  const baseDate = periodDate ? moment(periodDate).format(DATE_FORMAT) : moment().format(DATE_FORMAT);
+
+  const queryPeriodRange = `
+    SELECT id FROM period WHERE id BETWEEN ? AND ?;
+  `;
+
+  const queryPeriodId = periodId ?
+    'SELECT id FROM period WHERE id = ? LIMIT 1;' :
+    'SELECT id FROM period WHERE DATE(?) BETWEEN DATE(start_date) AND DATE(end_date) LIMIT 1;';
```

Comment:
If you want to make this bullet proof, it should use the Enterprise ID to make sure BHIMA works correctly with multiple enterprises  ... but I guess this is okay for now.